### PR TITLE
Update Content Security Policy

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
- Add missing docblocks.
- Fix phpstan ignored line.
- Fix formatting
- Fix IDE errors
 
**Checklist:**
- [x] Securely signed commits
- [x] Conforms to style guide

### CODE REVIEW

**Comment by paulbalandan** [comment]
Anyone to review? @samsonasik @MGatner ?

File: system/HTTP/ContentSecurityPolicy.php [hunk 1]
```diff
@@ -188,52 +185,47 @@ class ContentSecurityPolicy
...
 	 *
 	 * @param ContentSecurityPolicyConfig $config
 	 */
 	public function __construct(ContentSecurityPolicyConfig $config)
 	{
-		foreach ($config as $setting => $value) // @phpstan-ignore-line
+		foreach (get_object_vars($config) as $setting => $value)
 		{
-			if (isset($this->{$setting}))
+			if (property_exists($this, $setting))
```
Comment (samsonasik) [inline system/HTTP/ContentSecurityPolicy.php:199]: property_exists with isset is different, property_exists got data even value is null. ref https://3v4l.org/m99JJ

Comment (paulbalandan) [reply system/HTTP/ContentSecurityPolicy.php:199]: Yes. But isn't it a bug to use `isset`? For example, this class' `$reportURI` property is uninitialized so `isset` to it will evaluate to `false` and I cannot set it in the constructor.

Comment (samsonasik) [reply system/HTTP/ContentSecurityPolicy.php:199]: probably, @MGatner any opinion?

Comment (MGatner) [reply system/HTTP/ContentSecurityPolicy.php:199]: Yes let's use `property_exists` so we can override the null values.

File: system/HTTP/ContentSecurityPolicy.php [hunk 2]
```diff
@@ -624,17 +584,20 @@ protected function addOption($options, string $target, bool $explicitReporting =
...
-	 * @param ResponseInterface|Response $response
+	 * @param ResponseInterface $response
+	 *
+	 * @return void
 	 */
 	protected function generateNonces(ResponseInterface &$response)
 	{
+		/**
+		 * @var Response $response
+		 */
```
Comment (MGatner) [inline system/HTTP/ContentSecurityPolicy.php:600]: Why is this necessary over ResponseInterface? If someone is using their own Response class then this still needs to work.

**Review by MGatner — APPROVED** [review]
Very nice cleanup! Just the one incidental comment.

File: app/Config/ContentSecurityPolicy.php [hunk 3]
```diff
@@ -5,44 +5,155 @@
...
+	 */
+	public $formAction = 'self';
+
+	/**
+	 * Specifies the sources that can embed the current page.
+	 * This directive applies to `<frame>`, `<iframe>`, `<embed>`,
+	 * and `<applet>` tags. This directive can't be used in
+	 * `<meta>` tags and applies only to non-HTML resources.
+	 *
+	 * @var string|string[]
```
Comment (samsonasik) [inline app/Config/ContentSecurityPolicy.php:123]: @var string|string[]|null

File: system/HTTP/ContentSecurityPolicy.php (hunk 2 above)
Comment (paulbalandan) [reply system/HTTP/ContentSecurityPolicy.php:600]: I just put it there so that vscode will not turn red everytime I open this file. The current `ResponseInterface` does not have the method used there but is found in `Response`.

Comment (MGatner) [reply system/HTTP/ContentSecurityPolicy.php:600]: Hmm that seems to me a larger issue, that code framework code could rely on an incomplete interface. I would be in favor of considering the current ResponseInterface bugged by not accounting for `getBody()` and thus adding it. Thoughts for another Issue perhaps.

Comment (paulbalandan) [reply system/HTTP/ContentSecurityPolicy.php:600]: OK, I'll revert that for now.

**Review by samsonasik — APPROVED** [review]

**Comment by paulbalandan** [comment]
@samsonasik @MGatner thank you for the reviews
